packages/cozy-app-publish/lib/prepublish.js

```diff
@@ -71,30 +72,36 @@ const check = options => {
   return options
 }
 
-const shasum = options => {
-  const { appBuildUrl, verbose } = options
-  // get the sha256 hash from the archive from the url
-  const shaSumProcess = spawn.sync(
-    'sh',
-    ['-c', `curl -sSL --fail "${appBuildUrl}" | shasum -a 256 | cut -d" " -f1`],
-    {
-      stdio: verbose ? 'inherit' : 'pipe'
-    }
-  )
-  // Note: if the Url don't return an archive or if 404 Not found,
-  // the shasum will be the one of the error message from the curl command
-  // so no error throwed here whatever the url is
-  if (shaSumProcess.status !== 0) {
-    throw new Error(
-      `Error from archive shasum computing (${appBuildUrl}). Publishing failed.`
-    )
-  }
-  // remove also the ending line break
-  options.sha256Sum = shaSumProcess.stdout.toString().replace(/\r?\n|\r/g, '')
+const shasum256FromURL = url =>
+  new Promise((resolve, reject) => {
+    const hash = crypto.createHash('sha256')
+    const httpLib = url.indexOf('https://') === 0 ? https : http
+    const req = httpLib.get(url, res => {
+      res.on('data', d => {
+        hash.update(d)
+      })
+
+      res.on('end', () => {
+        resolve(hash.digest('hex'))
+      })
+    })
+
+    req.on('error', e => {
+      reject(e)
+    })
+  })
+
+const shasum = async options => {
+  const { appBuildUrl } = options
+  const shasum = await prepublish.shasum256FromURL(appBuildUrl)
```

Comment:
Nit: I would call directly `shasum256FromURL` here. Is there a reason to use `prepublish` ?

Reply:
This is to be able to mock it in tests.